common/lib/xmodule/xmodule/js/src/javascript_loader.coffee

```diff
@@ -66,5 +66,52 @@ class @JavascriptLoader
         # just call the completion callback directly, without reloading the file
         completionHandlerGenerator(index)()
       $(placeholder).remove()
-        
-      
+
+  # loadXBlockResources:
+  #   With the provided XBlock resources:
+  #     1) Explicitly attach the resource to the <head> of document
+  #     2) Explicitly wait for each resource to be loaded
+  #     3) Return to callback function when all resources loaded
+  @loadXBlockResources: (resources, callback=null) ->
```

Comment:
If it doesn't make sense to reuse all the logic from Studio, I think you should take this method instead of implementing it again:

https://github.com/edx/edx-platform/blob/master/cms/static/js/views/xblock.js#L177

In particular, we had trouble ensuring that the scripts had all completed loading before resolving the promise, so we ended up rolling our own helper method that uses the RequireJS loader which seems very robust.